pkg/controller/controller.go

```diff
@@ -331,14 +333,21 @@ func (cont *AciController) Init() {
 
 	cont.log.Debug("Initializing IPAM")
 	cont.initIpam()
-	//@FIXME need to set this value based on feature capability supported by cluster
+	// check if the cluster supports endpoint slices
 	// if cluster doesn't have the support fallback to endpoints
-	if cont.config.EnabledEndpointSlice {
+	endpointSlice := false
+	kubeClient := cont.env.(*K8sEnvironment).kubeClient
+	esobj, err := kubeClient.DiscoveryV1beta1().EndpointSlices("default").Get(context.TODO(), "kubernetes", metav1.GetOptions{})
+	if err == nil && esobj != nil {
```

Comment:
Can this logic be refactored so that it can be used commonly in the host agent as well (as opposed to duplicating the code)?

Reply:
This is single line to get the endpoint slice Obj, I guess it should be ok, reset of the code is initialize the  kubeClient is specific to controller or host agent

Reply:
let me move it to common this function